lib/plugin.js

```diff
@@ -75,19 +75,22 @@ ManifestPlugin.prototype.apply = function(compiler) {
       return memo;
     }, {}));
 
-    // Append optional basepath onto all references.
-    // This allows output path to be reflected in the manifest.
-    if (this.opts.basePath) {
+    // Prepend an optional public path onto all manifest values,
+    // similar to Webpack's `output.publicPath` option. Example:
+    // {"foo/bar.js":"foo/bar.123.js"} becomes {"foo/bar.js":"public/foo/bar.123.js"}
+    if (this.opts.publicPath) {
       manifest = _.reduce(manifest, function(memo, value, key) {
-        memo[resolvePath(this.opts.basePath, key)] = resolvePath(this.opts.basePath, value);
+        memo[key] = resolvePath(this.opts.publicPath, value);
         return memo;
       }.bind(this), {});
-    } else if (this.opts.publicPath) {
-      // Similar to basePath but only affects the value (similar to how
-      // output.publicPath turns require('foo/bar') into '/public/foo/bar', see
-      // https://github.com/webpack/docs/wiki/configuration#outputpublicpath
+    }
+
+    // Prepend an optional base path onto all manifest keys AND values.
+    // This allows Webpack's `output.path` to be reflected in the manifest. Example:
+    // {"foo/bar.js":"foo/bar.123.js"} becomes {"path/to/foo/bar.js":"path/to/foo/bar.123.js"}
+    if (this.opts.basePath) {
       manifest = _.reduce(manifest, function(memo, value, key) {
-        memo[key] = resolvePath(this.opts.publicPath, value);
+        memo[resolvePath(this.opts.basePath, key)] = resolvePath(this.opts.basePath, value);
```

Comment:
I understand that the current behavior of `basePath` is acting on both key and value.
I believe it should only modify the key. Thoughts ?